aws/resource_aws_lambda_function.go

```diff
@@ -600,6 +600,32 @@ func resourceAwsLambdaFunctionUpdate(d *schema.ResourceData, meta interface{}) e
 			configUpdate = true
 		}
 	}
+	if d.HasChange("vpc_config") {
+		vpcConfigRaw := d.Get("vpc_config").([]interface{})
+		vpcConfig, ok := vpcConfigRaw[0].(map[string]interface{})
+		if !ok {
+			return errors.New("vpc_config is <nil>")
+		}
+
+		if vpcConfig != nil {
```

Comment:
Without looking too much into the Terraform core, could this be something other than nil, like empty or so?
(Just wondering if the check is needed, since the check above guard us from nil structures, but may be missing something)